### ISSUE TITLE
Source maps support

### ISSUE THREAD

**Issue by judofyr** [somebee/imba#1]
Any plans on supporting source maps?

**Comment by somebee** [comment]
The lexer already marks all tokens with their original location (both char and line/col), as it is used for analysis / annotations and the highlighter. I have limited knowledge of source maps, but from what I understand it should be relatively easy to support.

**Comment by marvinroger** [comment]
This feature is definitely needed for any serious project. Debugging unfortunately happens so often!

**Comment by judofyr** [comment]
It's not that bad actually. The JavaScript it generates is rather clean and
easy to debug. But yes, it's definitely a feature we want to implement.

**Comment by marvinroger** [comment]
That's right, the generated JS is clean, but when your debugger tells you that there is an error at x line, you lost some time to find the corresponding imba line. :)

**Comment by somebee** [comment]
@marvinroger Preliminary support for sourcemaps has been added and released with v0.12.0. The sourcemap is embedded directly in the compiled files if you include `--source-map-inline` in imba watch/compile. Make sure the imba-file is accessible from the browser (correctly relative to the compiled file). I've tested it in Chrome and it seems to work well. Still a few mappings missing, but it should be pretty usable :)
